app.js

```diff
@@ -1,16 +1,52 @@
 var http = require('http');
+var search = require("./search.js");
+
 var port = process.env.PORT || 2345;
 
+// WARNING there is no sanity check on parameters parsing!
+
+var basePath = "/suggestions?q=";
+var latParam = "latitude=";
+var lonParam = "longitude=";
+
+// Server start
 module.exports = http.createServer(function (req, res) {
-  res.writeHead(404, {'Content-Type': 'text/plain'});
-
-  if (req.url.indexOf('/suggestions') === 0) {
-    res.end(JSON.stringify({
-      suggestions: []
-    }));
-  } else {
-    res.end();
-  }
-}).listen(port, '127.0.0.1');
-
-console.log('Server running at http://127.0.0.1:%d/suggestions', port);
\ No newline at end of file
+	try {
+		var urlChunks = req.url.split("&");
```

Comment:
use the module from node `querystring` ?

Reply:
Possibly `querystring` is a better choice, this is the part where my bad knowledge of node libraries comes in.
I did not want to import `url` as I thought it was overkill for this, but I was not aware of `querystring`